src/Forms/GridField/GridFieldDetailForm_ItemRequest.php

```diff
@@ -561,6 +561,8 @@ public function getEditLink($id)
             $id
         );
 
+        $link = $this->gridField->addRequestToURL($link);
+
         return $this->getStateManager()->addStateToURL($this->gridField, $link);
```

Comment:
Is there ever a situation where we _only_ want the state for the current gridfield _or_ the state from all other gridfields to be added?
If not, it might make sense for the gridfield method to include `$this->getStateManager()->addStateToURL($this, $link);` at the end of it, so that here we only need to call the one method instead of two.
Same with edit button.

Reply:
DONE